pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -1368,16 +1399,31 @@ func (gce *GCECloud) NodeAddresses(_ string) ([]api.NodeAddress, error) {
 	}, nil
 }
 
-func (gce *GCECloud) isCurrentInstance(instance string) bool {
-	return gce.instanceID == canonicalizeInstanceName(instance)
+// Uses metadata server to check if specified instanceID matches current machine's instanceID
```

Comment:
nit: start the comment with the name of the function, e.g. isCurrentInstance uses the metadata ...